prow/cmd/phaino/main.go

```diff
@@ -58,6 +59,18 @@ func gatherOptions() options {
 	return o
 }
 
+func validate(pj prowapi.ProwJob) error {
+	switch {
+	case pj.Kind != "ProwJob":
+		return fmt.Errorf("bad kind: %q", pj.Kind)
+	case pj.Spec.PodSpec == nil && pj.Spec.Agent != prowapi.KubernetesAgent:
```

Comment:
nit: This would allow a PJ that specifies a PodSpec, but has agent set to `knative-build`. It may be marginally cleaner to just check the agent in this switch case.
/shrug